### PR TITLE
Add CiliumNetworkPolicy for CRD install job

### PR DESCRIPTION
**Pull request by fhielpos**
Towards: https://github.com/giantswarm/giantswarm/issues/24400

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Comment by fhielpos** [comment]
Tested on Golem WC ✅

**Review by fiunchinho — APPROVED** [review]

File: helm/kyverno/templates/crd-install/crd-np.yaml [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+      {{- include "kyverno-stack.selectorLabels" . | nindent 6 }}
+  egress:
+    - toEntities:
+        - kube-apiserver
+      toPorts:
+        - ports:
+            - port: "443"
+        - ports:
+            - port: "6443" 
+  {{- else}}
```
Comment (stone-z) [inline helm/kyverno/templates/crd-install/crd-np.yaml:29]: We need to keep the old one as well, right? Or does the cilium completely replace this on cilium clusters?

Comment (fhielpos) [reply helm/kyverno/templates/crd-install/crd-np.yaml:29]: I think this is mainly a bug with Cilium and it shouldn't replace it. However, in the current state, if you have both the traffic will still be blocked.

Comment (stone-z) [reply helm/kyverno/templates/crd-install/crd-np.yaml:29]: But if you have only the cilium one it works? Neat

**Review by stone-z — APPROVED** [review]

File: helm/kyverno/templates/crd-install/crd-np.yaml (hunk 1 above)
Comment (fhielpos) [reply helm/kyverno/templates/crd-install/crd-np.yaml:29]: Exactly 🚀

**Pull request merged**